bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FtpFolderWatcherHandler.java

```diff
@@ -0,0 +1,263 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}, error message {}", currentFtpListingFile, e.getMessage());
+            return;
+        }
+        this.initJob = scheduler.scheduleWithFixedDelay(this::connectionKeepAlive, 0, config.pollInterval,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> executionJob = this.executionJob;
+        ScheduledFuture<?> initJob = this.initJob;
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+        if (initJob != null) {
+            initJob.cancel(true);
+        }
+        if (ftp.isConnected()) {
+            try {
+                ftp.logout();
+                ftp.disconnect();
+            } catch (IOException e) {
+                logger.debug("Error terminating FTP connection: ", e);
+            }
+        }
+    }
+
+    private List<String> listDirectory(FTPClient ftpClient, String parentDir, String currentDir, boolean recursive)
+            throws IOException {
+        Instant dateNow = Instant.now();
+        List<String> dirList = new ArrayList<>();
+        final String dirToList = parentDir + ((!currentDir.equals("")) ? ("/" + currentDir) : "");
+
+        for (FTPFile file : ftpClient.listFiles(dirToList)) {
+            String currentFileName = file.getName();
+            if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                continue;
+            }
+            if (file.isDirectory()) {
+                if (recursive) {
+                    try {
+                        dirList.addAll(listDirectory(ftpClient, dirToList, currentFileName, recursive));
+                    } catch (IOException e) {
+                        logger.debug("Can't read FTP directory: {}", dirToList, e);
+                    }
+                }
+            } else {
+                long diff = ChronoUnit.HOURS.between(file.getTimestamp().toInstant(), dateNow);
+
+                if (diff < config.diffHours) {
+                    dirList.add("ftp:/" + ftpClient.getRemoteAddress().toString() + dirToList + "/" + currentFileName);
+                }
+            }
+        }
+        return dirList;
+    }
+
+    private void connectionKeepAlive() {
+        if (!ftp.isConnected()) {
+            switch (config.secureMode) {
+                case "NONE":
+                    ftp = new FTPClient();
+                    break;
+                case "IMPLICIT":
+                    ftp = new FTPSClient(true);
+                    break;
+                case "EXPLICIT":
+                    ftp = new FTPSClient(false);
+                    break;
+            }
+
+            int reply = 0;
+            ftp.setListHiddenFiles(config.listHidden);
+            ftp.setConnectTimeout(config.connectionTimeout * 1000);
+
+            try {
+                if (config.ftpPort > 0) {
```

Comment:
the default is automatically used, so you can always `ftp.connect(config.ftpAddress, config.ftpPort);`